server/service/goalService.js

```diff
@@ -666,6 +666,36 @@ exports.getRecentGoalEnrollmentEvents = async (limit) => {
     }  
 }
 
+exports.deleteGoalById = async (goalId) => {
+    let text = "DELETE FROM goals WHERE id = $1";
+    let values = [goalId];
```

Comment:
To implement the check for ownership, we should modify to something like the following:

```suggestion
exports.deleteGoalById = async (goalId, ownerId) => {
    let text = "DELETE FROM goals WHERE id = $1 and owned_by = $2";
    let values = [goalId, ownerId];
```